lib/faraday/response/raise_error.rb

```diff
@@ -28,6 +28,8 @@ def on_complete(env)
           raise Faraday::ConflictError, response_values(env)
         when 422
           raise Faraday::UnprocessableEntityError, response_values(env)
+        when nil
+          raise Faraday::NilStatusError, response: response_values(env)
```

Comment:
1. I'd rather have the `nil` check at the bottom, since it's the least likely to occur than any numerical status. Pretty minor issue, so I'm not sure if it's worth the effort :)
2. I like passing `response` as a named variable, rather than positional. I've added a note to the v2 wishlist (#953) to make this consistent with the other error types.